update tree output for exec to make more readable

coryb opened the pull request:
Current output:
```
hlb run --tree ./source.hlb
[+] Building 1.6s (1/1) FINISHED
 => compiling [default]                                                                                                                                                                                        1.6s
.
└── parallel
    └── parallel
        ├── single
        │   └── [exec]  meta:<args:"/bin/sh" args:"-c" args:"v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS=\"-X github.com/openllb/hlb.Version=$v\" /cross/build \"github.com/openllb/hlb/cmd/hlb\"" env:"PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" env:"GOLANG_VERSION=1.13.10" env:"OSX_CROSS_PATH=/osxcross" env:"GOPATH=/root/go" env:"GO111MODULE=on" cwd:"/go/src/hlb" > mounts:<dest:"/" > mounts:<input:1 selector:"/scripts" dest:"/cross" output:-1 readonly:true > mounts:<input:-1 dest:"/go/pkg/mod" output:-1 mountType:CACHE cacheOpt:<ID:"hlb/go-mod" sharing:PRIVATE > > mounts:<input:2 dest:"/go/src/hlb" output:-1 readonly:true > mounts:<input:-1 dest:"/root/.cache/go-build" output:-1 mountType:CACHE cacheOpt:<ID:"hlb/go-build" sharing:PRIVATE > > mounts:<input:-1 dest:"/root/go/bin" output:1 >
        │       ├── [source]  identifier:"docker-image://docker.io/dockercore/golang-cross:1.13.10"
        │       ├── [source]  identifier:"git://github.com/hinshun/go-cross.git" attrs:<key:"git.fullurl" value:"https://github.com/hinshun/go-cross.git" >
        │       └── [source]  identifier:"local://sha256:b5a53a856aa2c608a534f9fa01fd455bacbcabd94d21021d225e0d51ba854a3d" attrs:<key:"local.excludepatterns" value:"[\"build\"]" > attrs:<key:"local.session" value:"hynv9j9mpslyliafpsgc9sec4" >
        └── single
            └── [exec]  meta:<args:"/bin/sh" args:"-c" args:"v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS=\"-X github.com/openllb/hlb.Version=$v\" /cross/build \"github.com/openllb/hlb/cmd/hlb\"" env:"PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" env:"GOLANG_VERSION=1.13.10" env:"OSX_CROSS_PATH=/osxcross" env:"GOPATH=/root/go" env:"GO111MODULE=on" cwd:"/go/src/hlb" > mounts:<dest:"/" > mounts:<input:1 selector:"/scripts" dest:"/cross" output:-1 readonly:true > mounts:<input:-1 dest:"/go/pkg/mod" output:-1 mountType:CACHE cacheOpt:<ID:"hlb/go-mod" sharing:PRIVATE > > mounts:<input:2 dest:"/go/src/hlb" output:-1 readonly:true > mounts:<input:-1 dest:"/root/.cache/go-build" output:-1 mountType:CACHE cacheOpt:<ID:"hlb/go-build" sharing:PRIVATE > > mounts:<input:-1 dest:"/root/go/bin" output:1 >
                ├── [source]  identifier:"docker-image://docker.io/dockercore/golang-cross:1.13.10"
                ├── [source]  identifier:"git://github.com/hinshun/go-cross.git" attrs:<key:"git.fullurl" value:"https://github.com/hinshun/go-cross.git" >
                └── [source]  identifier:"local://sha256:b5a53a856aa2c608a534f9fa01fd455bacbcabd94d21021d225e0d51ba854a3d" attrs:<key:"local.excludepatterns" value:"[\"build\"]" > attrs:<key:"local.session" value:"hynv9j9mpslyliafpsgc9sec4" >
```

Output with PR:
```
./hlb run --tree ./source.hlb
[+] Building 1.7s (1/1) FINISHED
 => compiling [default]                                                                                                                                                                                        1.7s
.
└── parallel
    └── parallel
        ├── single
        │   └── [exec]  v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS="-X github.com/openllb/hlb.Version=$v" /cross/build "github.com/openllb/hlb/cmd/hlb"
        │       ├── [env]  PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
        │       ├── [env]  GOLANG_VERSION=1.13.10
        │       ├── [env]  OSX_CROSS_PATH=/osxcross
        │       ├── [env]  GOPATH=/root/go
        │       ├── [env]  GO111MODULE=on
        │       ├── [cwd]  /go/src/hlb
        │       ├── [mount]  docker-image://docker.io/dockercore/golang-cross:1.13.10 => / [type=BIND]
        │       ├── [mount]  git://github.com/hinshun/go-cross.git/scripts => /cross [type=BIND,ro]
        │       ├── [mount]  scratch => /go/pkg/mod [type=CACHE,cache-id=hlb/go-mod,sharing=PRIVATE]
        │       ├── [mount]  local://sha256:4e682aa48bfe4517ae4f1372a4c72944a5d1bc6c8bf2466ac9cac4fc3c13a63b => /go/src/hlb [type=BIND,ro]
        │       ├── [mount]  scratch => /root/.cache/go-build [type=CACHE,cache-id=hlb/go-build,sharing=PRIVATE]
        │       └── [mount]  scratch => /root/go/bin [type=BIND]
        └── single
            └── [exec]  v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS="-X github.com/openllb/hlb.Version=$v" /cross/build "github.com/openllb/hlb/cmd/hlb"
                ├── [env]  PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
                ├── [env]  GOLANG_VERSION=1.13.10
                ├── [env]  OSX_CROSS_PATH=/osxcross
                ├── [env]  GOPATH=/root/go
                ├── [env]  GO111MODULE=on
                ├── [cwd]  /go/src/hlb
                ├── [mount]  docker-image://docker.io/dockercore/golang-cross:1.13.10 => / [type=BIND]
                ├── [mount]  git://github.com/hinshun/go-cross.git/scripts => /cross [type=BIND,ro]
                ├── [mount]  scratch => /go/pkg/mod [type=CACHE,cache-id=hlb/go-mod,sharing=PRIVATE]
                ├── [mount]  local://sha256:4e682aa48bfe4517ae4f1372a4c72944a5d1bc6c8bf2466ac9cac4fc3c13a63b => /go/src/hlb [type=BIND,ro]
                ├── [mount]  scratch => /root/.cache/go-build [type=CACHE,cache-id=hlb/go-build,sharing=PRIVATE]
                └── [mount]  scratch => /root/go/bin [type=BIND]
```

Note the `[source]` lines are merged with the corresponding `[mount]` lines. 

btw, there does seem to be a subtle bug in our graph generation, not sure why we are duping the same `single` twice, but will try to fix that in seperate PR.

hinshun commented:
We are duping the same `single` twice because one is a regular solve, and the other is a export to `./build`.

hinshun commented on solver/request.go:
```diff
@@ -98,24 +100,70 @@ func (o op) Tree(tree treeprint.Tree) error {
 
 	switch v := pbOp.Op.(type) {
-	case *pb.Op_Source:
-		branch = tree.AddMetaBranch("source", v.Source)
```
```sh
$ cat foo.hlb
fs foo() {
	image "alpine"
}

$ hlb run --tree -t foo foo.hlb
[+] Building 0.0s (1/1) FINISHED
 => compiling [foo]                                                                                                                                                   0.0s
.
└── parallel
    └── single
```

coryb replied:
fixed, now we see:
```
./hlb run --tree ./test.hlb
[+] Building 0.0s (1/1) FINISHED
 => compiling [default]                                                                                                                                                                                        0.0s
.
└── parallel
    └── single
        └── [source]  identifier:"docker-image://docker.io/library/alpine:latest"
```

coryb commented on solver/request.go:
```diff
@@ -98,24 +100,70 @@ func (o op) Tree(tree treeprint.Tree) error {
-			return err
-		}
-	}
-
```
fixed, now report Inputs not reported via mounts

coryb commented:
One more issue, I would like to resolve what `local://` means here:
```
├── [mount]  local://sha256:4e682aa48bfe4517ae4f1372a4c72944a5d1bc6c8bf2466ac9cac4fc3c13a63b => /go/src/hlb [type=BIND,ro]          
```
From the attrs I can see the include/excludePatterns, but we lose the input path `.` in this case.  Looking at the def, I dont see anywhere the input path is tracked.  One idea is that we could use the `llb.SharedKeyHint(string)` llb.LocalOption, but not sure what that is intended for.  My idea was to do something like:
```
hint := fmt.Sprintf("local://%s", path)
opts = append(opts, llb.SessionID(cg.sessionID), llb.SharedKeyHint(hint)
```
This seems to work at least, it will provide an attr with key `local.sharedKeyHint`.  Not sure if there is any other implications of this though.

It would allow me to generate something this in the tree:
```
├── [mount]  local://vendor,excludePatterns=build => /go/src/hlb [type=BIND,ro]          
```

hinshun commented:
The `SharedKeyHint` is a hint for the underlying file sync service to understand that multiple keys refer to the same directory.  I'm not sure what the repercussions are for just a generic hint like `.`. Perhaps its safe if they are on different sessions?

Anyway, I think it's best we don't use that and instead use `llb.WithDescription`. That ends up in the protobuf op as well.

coryb commented:
Okay, fixed the local output mapping, also removed the uncessary `single` branch, it is implicit in the 
tree form.  Also added `ignoreCache` indicators.

This is the latest output:
```
./hlb run --tree ./source.hlb
[+] Building 1.7s (1/1) FINISHED
 => compiling [default]                                                                                                                                                                                        1.7s
.
└── parallel
    └── parallel
        ├── [exec]  v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS="-X github.com/openllb/hlb.Version=$v" /cross/build "github.com/openllb/hlb/cmd/hlb"
        │   ├── [env]  PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
        │   ├── [env]  GOLANG_VERSION=1.13.10
        │   ├── [env]  OSX_CROSS_PATH=/osxcross
        │   ├── [env]  GOPATH=/root/go
        │   ├── [env]  GO111MODULE=on
        │   ├── [cwd]  /go/src/hlb
        │   ├── [mount]  docker-image://docker.io/dockercore/golang-cross:1.13.10 => / [type=BIND]
        │   ├── [mount]  git://github.com/hinshun/go-cross.git/scripts => /cross [type=BIND,ro]
        │   ├── [mount]  scratch => /go/pkg/mod [type=CACHE,cache-id=hlb/go-mod,sharing=PRIVATE]
        │   ├── [mount]  local://.,excludepatterns=["build"] => /go/src/hlb [type=BIND,ro]
        │   ├── [mount]  scratch => /root/.cache/go-build [type=CACHE,cache-id=hlb/go-build,sharing=PRIVATE]
        │   └── [mount]  scratch => /root/go/bin [type=BIND]
        └── [exec]  v=$(git describe --match 'v[0-9]*' --tags --dirty='.dirty' --always | sed 's/^v//') && LDFLAGS="-X github.com/openllb/hlb.Version=$v" /cross/build "github.com/openllb/hlb/cmd/hlb"
            ├── [env]  PATH=/osxcross/target/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
            ├── [env]  GOLANG_VERSION=1.13.10
            ├── [env]  OSX_CROSS_PATH=/osxcross
            ├── [env]  GOPATH=/root/go
            ├── [env]  GO111MODULE=on
            ├── [cwd]  /go/src/hlb
            ├── [mount]  docker-image://docker.io/dockercore/golang-cross:1.13.10 => / [type=BIND]
            ├── [mount]  git://github.com/hinshun/go-cross.git/scripts => /cross [type=BIND,ro]
            ├── [mount]  scratch => /go/pkg/mod [type=CACHE,cache-id=hlb/go-mod,sharing=PRIVATE]
            ├── [mount]  local://.,excludepatterns=["build"] => /go/src/hlb [type=BIND,ro]
            ├── [mount]  scratch => /root/.cache/go-build [type=CACHE,cache-id=hlb/go-build,sharing=PRIVATE]
            └── [mount]  scratch => /root/go/bin [type=BIND]
```
